taichi/backends/metal/taichi_file_util.h

```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+#include <optional>
+#include <string>
+#include <vector>
+
+#include "taichi/backends/metal/kernel_util.h"
+
+TLANG_NAMESPACE_BEGIN
+
+class Kernel;
+class SNode;
+
+namespace metal {
+struct BufferSize {
```

Comment:
Nit: `BufferMetadata` could be a slightly better naming, because this contains more than  buffer sizes?